models/enterprise_sonic/l2_acls/sonic_l2_acls.yml

```diff
@@ -0,0 +1,206 @@
+---
+GENERATOR_VERSION: '1.0'
+ANSIBLE_METADATA: |
+    {
+        'metadata_version': '1.1',
+        'status': ['preview'],
+        'supported_by': 'community',
+        'license': 'Apache 2.0'
+    }
+NETWORK_OS: sonic
+RESOURCE: l2_acls
+COPYRIGHT: Copyright 2022 Dell Inc. or its subsidiaries. All Rights Reserved
+DOCUMENTATION: |
+    module: sonic_l2_acls
+    version_added: '2.1.0'
+    short_description: Manage Layer 2 access control lists (ACL) configurations on SONiC
+    description:
+      - This module provides configuration management of Layer 2 access control lists (ACL)
+        in devices running SONiC.
+    author: 'Arun Saravanan Balachandran (@ArunSaravananBalachandran)'
+    options:
+      config:
+        description:
+          - Specifies Layer 2 ACL configurations.
+        type: list
+        elements: dict
+        suboptions:
+          name:
+            description:
+              - Specifies the ACL name.
+            type: str
+            required: true
+          remark:
+            description:
+              - Specifies remark for the ACL.
+            type: str
+          rules:
+            description:
+              - List of rules with the ACL.
+              - I(sequence_num), I(action), I(source) & I(destination) are required for adding a new rule.
+              - If I(state=deleted), options other than I(sequence_num) are not considered.
+            type: list
+            elements: dict
+            required_together: [[action, source, destination]]
```

Comment:
- When state is merged, replaced or overridden `action`, `source`, and `destination` are required along with `sequence_num` for adding a new rule.
- When state is deleted, only `sequence_num` is required for deleting a specific rule.

Since `sequence_num` is required for all the supported states, it is marked as required while other options are not.

Reply:
Thanks for the details.

Validation performed by 'required_if' is only on the options within that sub-spec and does not consider the parent options.
(ref: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/validation.py#L246-L325)

In this scenario, since `state` is an option in the parent of `rules` spec 'required_if' cannot validate the same.

Reply:
Yes. In retrospect, that should have been obvious. There is really no other way it could work. Thanks for providing the code link and additional explanation.

With the information you have provided regarding the requirements in the "description", users and coders will have adequate documentation of this requirement.

You can mark this issue "resolved".